app/models/forms/base.rb

```diff
@@ -0,0 +1,42 @@
+module Forms
+  class Base < YAAF::Form
+    after_save :destroy_associated_models
+    attr_reader :main_models, :related_modes, :associated_forms
+
+    delegate :to_param, :persisted?, to: :main_model
+
+    def initialize(attributes = {})
+      super(attributes)
+
+      @models = parent_models + [main_model] + associated_forms
```

Comment:
This is the crux of configuration for YAAF: providing the models to be persisted. Only the main model is required.

Reply:
P.S. The order matters here for active record.